scripts/gathering/connected_GenerateMetalympicRewardTx.ts

```diff
@@ -0,0 +1,60 @@
+import {ethers} from 'hardhat';
+import {read} from '../utils/spreadsheet';
+
+async function main() {
+  const provider = ethers.provider;
+  const AssetAddress = '0xa342f5d851e866e18ff98f351f2c6637f4478db5';
+  const AssetId =
+    '55464657044963196816950587289035428064568320970692304673817341489687556012034';
+
+  const data: Array<string[]> = await read(
+    {
+      document: '1aSFdWvHpQEQrUu6eRVRNfVan8-OVILS16ft6uowhX5c',
+      sheet: 'MetalympicReward',
+    },
+    'D3:D288'
+  );
+
+  const cleanedData: string[] = [];
+
+  data.map((val) => cleanedData.push(val[0]));
+
+  const ids: string[] = [];
+  const amounts: number[] = [];
+
+  const resolvedAdd = async (address: string) => {
+    if (!address.match(/^0x[a-fA-F0-9]{40}$/g)) {
+      return provider.resolveName(address);
+    }
+    return address;
+  };
+
+  const script = async () =>
```

Comment:
You probably need to break into batch of 100 as more than 100 addresses could lead to have a gas requirement higer than ethereum block gas limit

Reply:
seb did the transaction with metalympics and it works with 285 elements
when i told him he needed to do 16tx for the inspector gumshoe he told me it's a lot and will take him to approve 16 times and pay fees 16 times and ask if we could do 1000 elements per 1000

what should i do? should we try to find the maximum by trying to submit higher and higher numbers?

Reply:
Ok, what the tx hash for the 285 so we can see how it cost.
we could also estimate it as part of the script

Re approval, there is no need to approve it 16 times, just one time

Reply:
il me semble que c'est celle ci 
https://etherscan.io/tx/0x996f86ecbc135a7618d84c91ebe26cfcfa453aab9ddb7f3bdcd1558cda52ad55